### ISSUE TITLE
nested ifs causes parsing error in shellcheck (for a valid shell script)

### ISSUE THREAD

**Issue by odulac** [koalaman/shellcheck#1088]
#### For bugs
- Rule Id: SC1075 (thrown abusively for nested ifs), causing SC1072; SC1073; SC1061 and SC1062 errors
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
while true
do
  if foo
  then echo " foo "
  else if [ -r bar ]
         then echo " bar is readable "
         else echo " bas unreadable or even non-existent"
         fi
  fi
done

# note: The script is correct (and needed in several cases)
# I don't test 3 different cases at the same level, I test for 2 cases at the first level,
#     and the 2nd case then needs a test to differentiate 2 different sub-cases on a 2nd level, 
# hence the nested if;...;fi. 
# But shellcheck refuses the "else if". This then causes various errors as the 2nd "if" becomes invalid.

```

#### Here's what shellcheck currently says:

$ shellcheck myscript
 
Line 2:
while true
^-- SC1073: Couldn't parse this while loop.
 
Line 3:
do
^-- SC1061: Couldn't find 'done' for this 'do'.
 
Line 6:
  else if [ -r bar ]
  ^-- SC1075: Use 'elif' instead of 'else if' (or put 'if' on new line if nesting).
 
Line 10:
  fi
  ^-- SC1062: Expected 'done' matching previously mentioned 'do'.
    ^-- SC1072: Unexpected keyword/token. Fix any mentioned problems and try again.

$ 

#### Here's what I wanted or expected to see:

Line 5: seems to appear as an error, and not a warning. The "else if" is needed here, and I can't replace it with an "elsif".

In addition, as it "doesn't accept" the "else if":
  - it then fails to see that the next "fi" is the end of just the 2nd "if"
  - and sees it instead as the end of the first if. 
Therefore the last "fi" becomes unexpected and is causing the SC1072 ; SC1073; SC1061 & SC1062 errors.

I suggest:  changing the error on "else if->elsif" and instead give a warning "didn't you want an elsif instead?" , and continue parsing normally (ie, enter the nested if, then exit it on its "fi", and return to the first level if).

Thanks in advance for your work !

**Comment by odulac** [comment]
sorry for all the edits on the bug report: I forgot the "do" in my first attempt, and corrected in several steps the bug report... (mention of different errors)

**Comment by koalaman** [comment]
This has come up several times before, but I closed them as wontfix because 80% of the time the user intended them as `elif` (so the current behavior gave overall better suggestions).

I finally came around and made it correct though. The noobs will have to fix this warning before getting any other useful suggestions, and I updated the SC1073 generic parser warning to mention that fixing it will offer more checks.

Thanks!

**Comment by odulac** [comment]
Thank you for fixing this: I also appreciate that it now suggest to pass the nested if on the next line (and once done it says there are no issues).
Thanks a lot for your work and this really great tool !
